kyaml/yaml/rnode.go

```diff
@@ -248,6 +443,391 @@ func (rn *RNode) IsNilOrEmpty() bool {
 		IsYNodeZero(rn.YNode())
 }
 
+// AsOrDie converts a RNode to the desired typed object. ptr must
+// be a pointer to a typed object. It will panic if it encounters an error.
+func (rn *RNode) AsOrDie(ptr interface{}) {
+	if err := rn.As(ptr); err != nil {
+		panic(err)
+	}
+}
+
+// As converts a RNode to the desired typed object. ptr must be
+// a pointer to a typed object.
+func (rn *RNode) As(ptr interface{}) error {
+	if rn.IsNil() {
+		return ErrRNodeNotFound{}
+	}
+	if ptr == nil || reflect.ValueOf(ptr).Kind() != reflect.Ptr {
+		return fmt.Errorf("ptr must be a pointer to an object")
+	}
+
+	// The built-in types only have json tags. We can't simply do mv.Node().Decode(ptr),
+	// since it use the lowercased field name by default if no yaml tag is specified.
+	// This affects both k8s built-in types (e.g. appsv1.Deployment) and any types
+	// that depends on built-in types (e.g. metav1.ObjectMeta, corev1.PodTemplate).
+	// To work around it, we rely on the json tags. We first convert mv to json
+	// and then unmarshal it to ptr.
```

Comment:
Similar to my other comments, please use k8syaml instead of json. Since we have rnodes to start with, we can just stringify them and let k8syaml take it from there.